### PR TITLE
[dashboard] Fix PrebuildStatus rendering and updating (on PrebuildList)

### PR DESCRIPTION
**Pull request by geropl**
## Description
This PR:
 - does make sure we are using the same `PrebuildStatus` component everywhere
 - registers listeners for not-done prebuilds on the `PrebuildList` page

## Related Issue(s)

Fixes ENT-271
Fixes ENT-231

## How to test
 - [join my org](https://gpl-271-pb-ready.preview.gitpod-dev.com/orgs/join?inviteId=84e2e93b-f0b8-4c94-b50a-eb1eeb736c78)
 - [trigger a prebuild](https://gpl-271-pb-ready.preview.gitpod-dev.com/prebuilds) on [gpl/multiple-tasks-short](https://github.com/geropl/gitpod-test-repo/tree/gpl/multiple-tasks-short)
 - watch the prebuild list + the prebuild detail view in parallel, and note how they both update after ~1min :heavy_check_mark: 

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - gpl-271-pb-ready</li>
	<li><b>🔗 URL</b> - <a href="https://gpl-271-pb-ready.preview.gitpod-dev.com/workspaces" target="_blank">gpl-271-pb-ready.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
	<li><b>📦 Version</b> - gpl-271-pb-ready-gha.26326</li>
	<li><b>🗒️ Logs</b> - <a href="https://console.cloud.google.com/logs/query;query=jsonPayload.kubernetes.host%3D%22preview-gpl-271-pb-ready%22%0A%0A--%20Filter%20on%20service:%0A--%20jsonPayload.serviceContext.service%3D%22ws-manager-mk2%22%0A;duration=P1D?project=gitpod-dev-preview" target="_blank">GCP Logs Explorer</a></li>
</ul>

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [x] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [x] /werft preemptible
      Saves cost. Untick this only if you're really sure you need a non-preemtible machine.
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
- [ ] with-monitoring
</details>

/hold

### CODE REVIEW

File: components/dashboard/src/projects/prebuild-utils.tsx [hunk 1]
```diff
@@ -5,17 +5,50 @@
...
 
 import StatusDone from "../icons/StatusDone.svg";
 import StatusFailed from "../icons/StatusFailed.svg";
 import StatusCanceled from "../icons/StatusCanceled.svg";
 import StatusPaused from "../icons/StatusPaused.svg";
 import StatusRunning from "../icons/StatusRunning.svg";
 
-export const prebuildDisplayProps = (prebuild: Prebuild): { className: string; label: string } => {
-    switch (prebuild.status?.phase?.name) {
+export function PrebuildStatus(p: { prebuild: Prebuild | undefined; classname?: string }) {
```
[the remark this answers is not in the document]
Comment (filiptronicek) [reply components/dashboard/src/projects/prebuild-utils.tsx:18]: nit: we could name the props `props` to make it consistent with the functions below, or even ditch the argument name altogether and not worry about it.

**Review by filiptronicek — APPROVED** [review]
Everything works as expected in all pages changed, including the cool new streaming under the Prebuilds list. Left a couple of comments.

File: components/dashboard/src/prebuilds/list/PrebuildList.tsx [hunk 2]
```diff
@@ -119,10 +123,38 @@ const PrebuildsListPage: FC = () => {
...
+
+            return watchPrebuild(prebuild.id, (update) => {
+                const index = prebuilds.findIndex((p) => p.id === prebuild.id);
+                if (index === -1) {
+                    console.warn("Can't handle prebuild update");
+                    return false;
+                }
+
+                prebuilds.splice(index, 1, update);
+                setPrebuilds([...prebuilds]);
```
[the remark this answers is not in the document]
Comment (geropl) [reply components/dashboard/src/prebuilds/list/PrebuildList.tsx:146]: react requires a new array object to detect something has changed :+1:

File: components/dashboard/src/prebuilds/detail/PrebuildDetailPage.tsx [hunk 3]
```diff
@@ -270,19 +240,11 @@ export const PrebuildDetailPage: FC = () => {
                                 </div>
                             </div>
                             <div className="flex flex-col gap-1 border-pk-border-base">
-                                <div className="py-4 flex flex-col gap-1">
-                                    <div className="px-6 flex gap-1 items-center">
-                                        {prebuildPhase.icon}
-                                        <span className="capitalize">{prebuildPhase.description}</span>{" "}
-                                        {isStreamingLogs && (
-                                            <TextMuted>
-                                                <MiddleDot /> Fetching logs...
```
[the remark this answers is not in the document]
Comment (geropl) [reply components/dashboard/src/prebuilds/detail/PrebuildDetailPage.tsx:279]: Let's try to make this a per-tab thing with the multi-tab approach.

**Pull request merged**